collector/pkg/metadata/kubernetes/pod_watch.go

```diff
@@ -44,6 +82,11 @@ func (m *podMap) add(info *K8sPodInfo) {
 	}
 	podInfoMap[info.PodName] = info
 	m.Info[info.Namespace] = podInfoMap
+	globalWorkload.add(&model.WorkloadInfo{
+		Namespace:    info.Namespace,
+		WorkloadName: info.WorkloadName,
+		WorkloadKind: info.WorkloadKind,
+	})
```

Comment:
Move `globalWorkload.add` out of the current scope. Because `podMap.add()` is only designed for adding `PodInfo` into the map, it should have a single responsibility.